sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CoalesceShufflePartitions.scala

```diff
@@ -66,12 +68,16 @@ case class CoalesceShufflePartitions(conf: SQLConf) extends Rule[SparkPlan] {
       val distinctNumPreShufflePartitions =
         validMetrics.map(stats => stats.bytesByPartitionId.length).distinct
       if (validMetrics.nonEmpty && distinctNumPreShufflePartitions.length == 1) {
+        // To avoid perf regression, we by default set the minimum number of coalesced partitions
```

Comment:
nit: We fall back to Spark default parallelism if the minimum number of coalesced partition is not set, so to avoid regressions compared to no coalescing.